### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by noctera**
| Status  | Type  |
| :---: | :---: |
| ✔️  Ready| Feature |

## Description

This PR will add the dashboard for the learning screen of the app.

It will show you all language packages, including any necessary information about your queries.

I used a route wrapper "Learn.jsx" (same principle like in Library) to easily move between dashboard, direction-, and learning screen.

Not working yet:
- redirects to direction screens (not existing yet)
- counter for "today" vocabularies

## How Has This Been Tested?

Locally

## Screenshots / GIFs (if appropriate):

dashboard

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking change which cleans up / improves existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the **CONTRIBUTING** document.
- [x] I have considered the accessibility of my changes (i.e. did I add proper content descriptions to images, or run my changes with talkback enabled?)
- [x] I have documented my code if needed

### CODE REVIEW

File: src/Components/PackageOverview/PackageOverview.scss [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    margin-left: 15px;
+  }
+
+  .package-unactivated {
+    grid-area: unactivated;
+    color: $color-main-text-inverse;
+    font-size: 15px;
+    margin: auto;
+    margin-left: 15px;
+  }
```
Comment (zikowang) [inline src/Components/PackageOverview/PackageOverview.scss:49]: We could shorten the style here:
```suggestion
  h1,
  p {
    color: $color-main-text-inverse;
    font-size: 15px;
    margin: auto;
    margin-left: 15px;

    &.package-heading {
      grid-area: heading;
      font-size: 25px;
    }

    &.package-unresolved {
      grid-area: unresolved;
    }

    &.package-today {
      grid-area: today;
    }

    &.package-unactivated {
      grid-area: unactivated;
    }
  }
```

File: src/screens/Learn/Dashboard/Dashboard.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      });
+  }, [showSnack]);
+
+  return (
+    <div className="dashboard">
+      <div className="package-views">
+        {languagePackages.map((languagePackage, index) => (
+          <PackageOverview key={index} data={languagePackage} />
+        ))}
+      </div>
```
Comment (zikowang) [inline src/screens/Learn/Dashboard/Dashboard.jsx:42]: Do we need the wrapper here?

**Review by zikowang — CHANGES_REQUESTED** [review]

File: src/screens/Learn/Dashboard/Dashboard.jsx (hunk 2 above)
Comment (noctera) [reply src/screens/Learn/Dashboard/Dashboard.jsx:42]: Not especially. I could also put some padding to the top of dashboard.

**Review by zikowang — APPROVED** [review]

File: src/Components/DirectionBox/DirectionBox.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import ArrowRightAltIcon from "@material-ui/icons/ArrowRightAlt";
+import SyncAltIcon from "@material-ui/icons/SyncAlt";
+
+import { languages } from "../../utils/constants.js";
+
+import "./DirectionBox.scss";
+
+const DirectionBox = ({ direction = "default" }) => {
+  const foreignWordLanguage = useSelector(
```
Comment (luwol03) [inline src/Components/DirectionBox/DirectionBox.jsx:12]: I think it's better if we use redux some levels down so that the component can be used stateles in other screens. The best way to transfer the values I think is via the props.

File: src/i18n/locales/en/electron.json [hunk 4]
```diff
@@ -11,4 +11,4 @@
       "restarting": "Restarting..."
     }
   }
-}
```
Comment (luwol03) [inline src/i18n/locales/en/electron.json:14]: can you undo this change?

File: src/utils/constants.js [hunk 5]
```diff
@@ -1,11 +1,111 @@
...
-  { id: 2, name: "Spanisch", icon: "🇪🇸" },
-  { id: 3, name: "Portugisisch", icon: "🇵🇹" },
-  { id: 4, name: "Türkisch", icon: "🇹🇷" },
-  { id: 5, name: "Russisch", icon: "🇷🇺" },
-  { id: 6, name: "Französisch", icon: "🇫🇷" },
+  {
+    id: 0,
+    name: "Deutsch",
+    icon: (
+      <ReactCountryFlag
```
Comment (luwol03) [inline src/utils/constants.js:8]: - id isn't necessary, we can just use the index of the element in the array
- can we use an array with two props and then map this?
```js
export const languages = [
  { name: "Deutsch", countryCode: "de" },
  ...
].map((language) => ({
  ...language,
  icon: (
      <ReactCountryFlag
        className="flag"
        style={{
          width: null,
          height: null,
        }}
        countryCode={language.countryCode}
        svg
      />
    ),
}))
```

**Review by luwol03 — CHANGES_REQUESTED** [review]

File: src/i18n/locales/en/electron.json (hunk 4 above)
Comment (noctera) [reply src/i18n/locales/en/electron.json:14]: Somehow I couldn't find this deletion in any of the branches

File: src/utils/constants.js (hunk 5 above)
Comment (noctera) [reply src/utils/constants.js]: done

**Review by zikowang — DISMISSED** [review]

**Review by luwol03 — APPROVED** [review]

**Review by zikowang — APPROVED** [review]
